src/firefly/java/edu/caltech/ipac/visualize/plot/RangeValues.java

```diff
@@ -50,170 +35,77 @@
-
-    public RangeValues(int algorithm ) {
-
-        this( PERCENTAGE, 1.0, PERCENTAGE, 99.0, ASINH_Q, GAMMA, algorithm, 25, 600, 120);
-    }
-    public RangeValues( int    lowerWhich,
-                        double lowerValue,
-                        int    upperWhich,
-                        double upperValue,
-                        int    algorithm) {
-        this( lowerWhich, lowerValue, upperWhich, upperValue, ASINH_Q, GAMMA,algorithm, 25, 600, 120);
+        this( PERCENTAGE, 1.0, PERCENTAGE, 99.0, ASINH_Q, GAMMA, STRETCH_LINEAR, 25, 600, 120, RGB_PRESERVE_HUE_DEFAULT);
     }
 
-    //added
     public RangeValues( int    lowerWhich,
                         double lowerValue,
                         int    upperWhich,
                         double upperValue,
                         double asinhQValue,
-                        double gammaValue,
-                        int    algorithm) {
-        this( lowerWhich, lowerValue, upperWhich, upperValue, asinhQValue, gammaValue, algorithm, 25, 600, 120);
-    }
-
-    public RangeValues( int    lowerWhich,
-                        double lowerValue,
-                        int    upperWhich,
-                        double upperValue,
+                        double gammaOrStretch,
                         int    algorithm,
                         int    zscale_contrast,
                         int    zscale_samples,
-                        int    zscale_samples_per_line) {
-
-        this( lowerWhich, lowerValue, upperWhich, upperValue, ASINH_Q, GAMMA, algorithm,
-                zscale_contrast, zscale_samples, zscale_samples_per_line,
-                0.5, 1.0);
-    }
-   //LZ added
-    public RangeValues( int    lowerWhich,
-                        double lowerValue,
-                        int    upperWhich,
-                        double upperValue,
-                        double asinhQValue,
-                        double gammaValue,
-                        int    algorithm,
-                        int    zscale_contrast,
-                        int    zscale_samples,
-                        int    zscale_samples_per_line) {
-
-        this( lowerWhich, lowerValue, upperWhich, upperValue, asinhQValue, gammaValue, algorithm,
-                zscale_contrast, zscale_samples, zscale_samples_per_line,
+                        int    zscale_samples_per_line,
+                        short  rgbPreserveHue) {
+        this( lowerWhich, lowerValue, upperWhich, upperValue, asinhQValue, gammaOrStretch, algorithm,
+                zscale_contrast, zscale_samples, zscale_samples_per_line, rgbPreserveHue,
                 0.5, 1.0);
     }
 
 
-    //LZ  added
     public RangeValues( int    lowerWhich,
                         double lowerValue,
                         int    upperWhich,
                         double upperValue,
                         double asinhQValue,
-                        double gammaValue,
+                        double gammaOrStretch,
```

Comment:
> gammaOrStretch in RangeValues
I was trying to reduce the number of passed parameters, because we pass RangeValues to the server and back all the time. We do not use 'gamma' for hue-preserving stretch, and do not use 'stretch' if it's not hue-preserving.  May be it's over-optimization. 11 or 12 parameters is not a big difference. I'll change it to separate parameters. It will be more straight-forward.

> I am feeling unsure about limiting stretch to 0.1. Is stretch in flux units? I'm wondering about the case where an image has small values.
The stretch is in flux units. I'll change it to 1e-10.